modules/shared-utils/src/cljdoc/util.clj

```diff
@@ -36,6 +37,28 @@
   {:pre [(some? project) (string? version)]}
   (str "cljdoc-edn/" (group-id project) "/" (artifact-id project) "/" version "/cljdoc.edn"))
 
+(defn- wrap-regex-tag [potential-regex]
+  (if (instance? java.util.regex.Pattern potential-regex)
+    (tagged-literal 'regex (str potential-regex))
+    potential-regex))
+
+(defn- wrap-all-regex-tags [structure]
+  (clojure.walk/postwalk wrap-regex-tag structure))
+
+(defn- unwrap-regex-tag [regex-tagged-string]
+  (re-pattern regex-tagged-string))
+
+(defn serialize-cljdoc-edn [analyze-result]
+  ;; the analyzed structure can contain regex #"..." (e.g. in :arglists)
+  ;; and they can't be read in again using edn/read-string
+  ;; so there are changed to #regex"..." and read in with a custom reader
+  (-> analyze-result
+      (wrap-all-regex-tags)
```

Comment:
IMHO having the code spread across multiple functions makes it slightly harder to read than if it were all in one function. How would you feel about this? 

```clojure
(->> analyze-result
     (clojure.walk/postwalk #(if (instance? java.util.regex.Pattern %)
                               (tagged-literal 'regex (str %))
                               %)))
     (pr-str))
```

Reply:
Usally when threading data, I tend to make all the steps into separate functions as it can be read more fluently top to bottom without figuring out what every single call exactly does (athough most of the time finding good names is the hard part 🙂). Having only one step, I'm totally fine with not having a separate function 🙂 and will change the PR following your suggestion.